### PR TITLE
Move global DSL type definitions into an `index` file.

### PR DESCRIPTION
**Pull request by philcockfield**
Export the UIHarness DSL as global types.

### CODE REVIEW

**Comment by philcockfield** [comment]
Hey @frederickfogerty - figuring out how to move the Type Definitions you declared in the `typescript.spec.tsx` file into a place that they are globally picked up by the parent module.

I've copied them out into a `/src/index.ts` file, but can't seem to get them to get picked up globally by the TS compiler.

Any ideas?

**Comment by frederickfogerty** [comment]
Did you mean to make it index.d.ts, not index.ts?

File: package.json [hunk 1]
```diff
@@ -76,7 +80,7 @@
     "sinon": "^1.16.1",
     "tslint": "^3.11.0",
     "tslint-react": "^0.4.0",
-    "typescript": "^1.9.0-dev.20160617-1.0",
+    "typescript": "^2.1.0-dev.20160716",
```
Comment (philcockfield) [inline package.json]: Sweet - new version.  That has `async/await` I assume.  Will update my other references to this too.

File: src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+export declare var describe: UIHarness.IBDDFunction;
+export declare var before: UIHarness.IBDDFunction;
+export declare var section: UIHarness.IBDDFunction;
+export declare var it: UIHarness.IBDDFunction;
+
+declare var defaultExport: { start: any, build: any };
+export default defaultExport;
+
```
Comment (philcockfield) [inline src/index.d.ts]: Freak'in BEAUTIFUL!!!! 🎱

**Comment by philcockfield** [comment]
Hey @frederickfogerty, is this the kind of error you were seeing?

image

I blew away `node_modules` and re-installed, and it still did this.  Any ideas?

**Comment by frederickfogerty** [comment]
@philcockfield nope, wasn't getting those errors - I wasn't getting any errors at all when running the ui-harness is dev mode

File: package.json (hunk 1 above)
Comment (frederickfogerty) [inline package.json]: Nope, still open https://github.com/Microsoft/TypeScript/pull/9175

**Pull request closed without merging**